### PR TITLE
gas optimizations

### PR DESCRIPTION
**Pull request by KillariDev**

### CODE REVIEW

File: solidity/contracts/PetalLock.sol [hunk 1]
```diff
@@ -108,16 +110,20 @@ contract PetalLock {
...
 
 	function makeImmutable(address originalOwner, SubDomainLabelNode[] memory pathToChild, bytes memory contenthash, address resolutionAddress) private {
+		uint256 pathToChildLength = pathToChild.length;
+		require(pathToChildLength > 0, 'PetalLock: Missing subdomain path');
 		require(ensRegistry.recordExists(pathToChild[0].node), 'PetalLock: The second level domain does not exist');
 		require(ensTokenWrapper.isWrapped(pathToChild[0].node), 'PetalLock: The second level domain is not wrapped');
-		uint256 finalChildIndex = pathToChild.length - 1;
+
+		uint256 finalChildIndex;
+		unchecked { finalChildIndex = pathToChildLength - 1; }
```
Comment (MicahZoltu) [inline solidity/contracts/PetalLock.sol]: I don't think the gas benefits are worth it for this.  The only time I think `unchecked` blocks are worth the gas benefits are when you have it in a tight loop.  In this case, it is called once so we are talking about a single or low double digit gas savings, but comes with a high auditability cost.

Comment (MicahZoltu) [reply solidity/contracts/PetalLock.sol]: Same comment for anywhere else `unchecked` is used.

**Review by MicahZoltu — APPROVED** [review]
Approved, but would really like to see the `unchecked` code removed before merging, or a much longer discussion about its benefits/costs.

File: solidity/contracts/PetalLock.sol (hunk 1 above)
Comment (KillariDev) [reply solidity/contracts/PetalLock.sol]: removed them. Yeah, they are all one time things

**Pull request merged**